fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -353,10 +352,10 @@ protected String buildWhereClause(String overrideType) {
 
                     whereClauseSegment = querySegment.getQueryString();
                     if (Modifier.MISSING.equals(param.getModifier())) {
-                        whereClause.append(AND).append(whereClauseSegment);
+                        whereClause.append(whereClauseSegment);
                     } else {
 
-                        whereClause.append(AND).append("R.LOGICAL_RESOURCE_ID IN (SELECT LOGICAL_RESOURCE_ID FROM ");
+                        whereClause.append(JOIN).append("(SELECT DISTINCT LOGICAL_RESOURCE_ID FROM ");
```

Comment:
I think adding a comment on 357 explaining the design choice for distinct to eliminate duplication is good.

Reply:
good point, will add